app/controllers/spree/admin/products_controller_decorator.rb

```diff
@@ -9,11 +9,10 @@ def update
     end
 
     #this part is added
-    if params[:product][:published].present?
-      params[:product][:published_at] = DateTime.current
+
+    if params[:product][:published].eql?('1')
       params[:product][:publisher_id] = spree_current_user.id
     else
-      params[:product][:published_at] = nil
```

Comment:
@gurpreet013 Instead of using params value, can we use association to assign the same?